templates/docs/examples/patterns/navigation/_script-sliding.js

```diff
@@ -106,30 +106,46 @@ const initNavigationSliding = () => {
   toggles.forEach(function (toggle) {
     toggle.addEventListener('click', function (e) {
       e.preventDefault();
+      closeSearch();
       const target = document.getElementById(toggle.getAttribute('aria-controls'));
       if (target) {
-        const isNested = target.parentNode.parentNode.classList.contains('p-navigation__dropdown');
-
+        const isNested = !target.closest('.p-navigation__dropdown');
         if (!isNested) {
           resetToggles(target);
         }
         if (target.getAttribute('aria-hidden') === 'true') {
           toggle.parentNode.classList.add('is-active');
           toggle.parentNode.parentNode.classList.add('is-active');
           target.setAttribute('aria-hidden', 'false');
+
+          // only animate the dropdown if menu is not open, otherwise just switch the visible one
+          if (!navigation.classList.contains('has-menu-open')) {
+            // trigger the CSS transition
+            requestAnimationFrame(() => {
+              target.classList.remove('is-collapsed');
+            });
+          } else {
+            // make it appear immediately
+            target.classList.remove('is-collapsed');
+          }
+          navigation.classList.add('has-menu-open');
           setFocusable(target);
         } else {
-          target.setAttribute('aria-hidden', 'true');
-          toggle.parentNode.classList.remove('is-active');
-          toggle.parentNode.parentNode.classList.remove('is-active');
+          target.classList.add('is-collapsed');
+          setTimeout(() => {
+            target.setAttribute('aria-hidden', 'true');
+            toggle.parentNode.classList.remove('is-active');
+            toggle.parentNode.parentNode.classList.remove('is-active');
+            navigation.classList.remove('has-menu-open');
+          }, 100);
```

Comment:
Yes, this actually should align with the transition time from CSS. But because we don't have a "JavaScript library" exported with Vanilla it's not standarised in JS in any way. But I can make it a const for the purposes of this script for clarity.